crates/core/src/client/mod.rs

```diff
@@ -635,6 +653,9 @@ impl<P: Provider + Send + Sync + 'static> KakarotStarknetApi<P> for KakarotClien
     ) -> Result<H256, EthApiError<P::Error>> {
         let transaction_result =
             self.starknet_provider.add_invoke_transaction(&BroadcastedInvokeTransaction::V1(request)).await?;
+        let waiter =
+            TransactionWaiter::new(self.starknet_provider(), transaction_result.transaction_hash, 1000, 15_000);
```

Comment:
is this the standard name for a struct or class that enables wait for tx?

Reply:
This is the name used by [dojo](https://github.com/dojoengine/dojo/blob/main/crates/dojo-world/src/utils.rs#L33).